### DISCUSSION TITLE
alphanumeric sort

### DISCUSSION THREAD

**Discussion by zakaria-chahboun** [supabase/supabase#3008] [Questions] [answered]
How i can do an alphanumeric order? i have names like this:
* test_1
* test_2
*  ...
* test_10
* test_11
* ....

i did this code:
```js
      let { data, error } = await supabase
        .from("test")
        .select("test_name")
        .order("test_name", { ascending: true});
```
Returns these results:
* test_1
* test_10
* test_11
* test_12
* test_13
* ....
* test_2
* test_3
* ....

But I expected:
* test_1
* test_2
*  ...
* test_10
* test_11
* ....

I found this solution in stackoverflow for prestgres sql
```sql
SELECT name
FROM folders
ORDER BY SUBSTRING(name FROM '([0-9]+)')::BIGINT ASC, name;
```
https://stackoverflow.com/a/32391788

How i can do like this answer in supabase in Javascript / RestAPI Url?

**Comment by soedirgo** [comment, marked as answer]
You can use [computed columns](https://postgrest.org/en/stable/api.html#computed-columns) for this, see e.g. https://github.com/supabase/supabase/discussions/2402. So it'll be something like:
```sql
CREATE FUNCTION test_number(t test) RETURNS bigint AS $$
  SELECT substring(t.test_name FROM '([0-9]+)')::bigint;
$$ LANGUAGE sql;
```
```js
const { data, error } = await supabase
    .from("test")
    .select("test_name")
    .order("test_number", { ascending: true });
```

**Reply by zakaria-chahboun** [reply]
Thanks
